beep/features/featurizer_helpers.py

```diff
@@ -633,21 +633,27 @@ def get_v_diff(processed_cycler_run, diag_pos, soc_window, baseline_step_index=1
     chosen_1 = chosen_1.loc[chosen_1.discharge_capacity.notna()]
     chosen_2 = chosen_2.loc[chosen_2.discharge_capacity.notna()]
 
+    # Filter so that only comparing on the same interpolation
+    chosen_2 = chosen_2[(chosen_1.discharge_capacity.min() < chosen_2.discharge_capacity) &
+                        (chosen_1.discharge_capacity.max() > chosen_2.discharge_capacity)]
+
     V = chosen_1.voltage.values
     Q = chosen_1.discharge_capacity.values
 
-    #  Round values so that non-strictly monotonic values are removed
-    np.around(Q, decimals=6, out=Q)
-    np.around(V, decimals=6, out=V)
-    if not np.all(np.diff(Q) > 0):
-        index_of_repeated = np.where(np.abs(np.diff(Q)) == 0)[0]
+    #  Threshold between values so that non-strictly monotonic values are removed
+    d_capacity_min = (np.max(Q) - np.min(Q)) / 1e6
```

Comment:
where does the 1e6 originate from? Something to do with the instrument resolution?

Reply:
It's just a threshold for the minimum resolution, i.e the maximum number of data points that we should use is 1e6. This is somewhat arbitrary, but does link back to the overall precision of the instrument (most ADC's are 24 bit). The problem here is somewhat sloppy coding, and some under-documented features of interp1d (the interpolation axis for the function does not have the same precision required for all modes)